### DISCUSSION TITLE
[ Best practice ] Background tasks and large operations

### DISCUSSION THREAD

**Discussion by Sharaf5** [emmett-framework/granian#348] [Q&A] [answered]
I wan't successful with celery along with other framworks
Or I didn't use it in a real live production environment

What is the best practice to implement background tasks with granian ?
and what it's considerations in production and dev ?

**Comment by gi0baro** [comment, marked as answer]
There's no implications in using Granian and dealing with background tasks; the requirements for that would be directly dependant on your application/the framework used.

The *warning* points are the same for every other server, like don't blocking the event loop in async code.

**Reply by Sharaf5** [reply]
I'm just testing Granian by itself for now and it seams promissing
so any documentation ? on how to handle any kind of either queued tasks or background long tasks
or any helpful resources ?

**Reply by gi0baro** [reply]
@Sharaf5 again, Granian is just a server, not a web framework. It has no facilities for *background tasks*.

**Reply by Sharaf5** [reply]
Then I should condsider coding or using it along with another framwork or module like celery or just go for threading
could you recommend one of three

**Reply by gi0baro** [reply]
I can't really make any suggestion without the project scope.
